docs/API-functions.md

```diff
@@ -0,0 +1,966 @@
+```
+// perfmon with Byte counters for better bitrate estimation.
+int srt_bstats(SRTSOCKET u, SRT_TRACEBSTATS * perf, int clear);
+
+// permon with Byte counters and instantaneous stats instead of moving averages for Snd/Rcvbuffer sizes.
+int srt_bistats(SRTSOCKET u, SRT_TRACEBSTATS * perf, int clear, int instantaneous);
+```
+
+Reports the current statistics
+
+* `u`: Socket to get stats from
+* `perf`: Pointer to an object to be written with the statistics
+* `clear`: 1 if the statis should be cleared after retrieval
+* `instantaneous`: 1 if the statistics should use instant data, not moving averages
+
+The `SRT_TRACEBSTATS` is an alias to `struct CBytePerfMon`. The meaning of most
+of the field should be enough comprehensible in the header file comments. Here
+are some less obvious fields in this structure (instant measurements):
+
+* `usPktSndPeriod`: sending period. This is the minimum time that must be kept
+between two consecutively sent packets over the link used by this socket (note
+that sockets sharing one outgoing port use the same underlying UDP socket and
+therefore the same link and the same sender queue). In other word, this is the
+inversion of maximum sending speed. This isn't the EXACT time distance between
+two consecutive sendings because in case when the time spent by the application
+between two consecutive sendings exceeds this time, then simply the next packet
+will be sent immediately, and additionally the extra wasted time will be
+"repaid" at the next sending.
+
+* `pktFlowWindow`: The "flow window" size, it's actually the number of free space
+in the peer receiver, as read on the sender, in the number of packets. When this
+value drops to zero, the next sending packet will be simply dropped by the receiver
+without processing. In the file mode this may cause slowdown of sending in order
+to wait until the receiver clears things up; in live mode the receiver buffer
+should normally occupy not more than half of the buffer size (default 8192).
+If this size is less than this half and declines, it means that the receiver
+cannot process the incoming stream fast enough and this may in perspective lead
+to a dropped connection.
+
+* `pktCongestionWindow`: The "congestion window" in packets. In File mode this
+value starts with 16 and is increased with every number of reported
+acknowledged packets, then also updated basing on the receiver-reported
+delivery rate. It represents the maximum number of packets that can be safely
+sent now without causing congestion. The higher this value, the faster the
+packets can be sent. In Live mode this field is not really in use.
+
+* `pktFlightSize`: Number of packets in flight. This is the distance between
+the packet sequence number that was last reported by ACK message and the
+sequence number of the packet just sent. Note that ACK gets received
+periodically, so this value is most accurate just after receiving ACK and
+becomes a little exaggerated in time until the next ACK comes.
```

Comment:
Research may be necessary to find out if this value shouldn't be (or even isn't) calculated as the distance between the packet that was last ACK-ed and the packet that was the last sent sequence *at the same time when receiving the ACK* (not at the time when the query was made).